website/source/api/secret/totp/index.html.md

```diff
@@ -0,0 +1,272 @@
+---
+layout: "api"
+page_title: "TOTP Secret Backend - HTTP API"
+sidebar_current: "docs-http-secret-totp"
+description: |-
+  This is the API documentation for the Vault TOTP secret backend.
+---
+
+# TOTP Secret Backend HTTP API
+
+This is the API documentation for the Vault TOTP secret backend. For
+general information about the usage and operation of the TOTP backend,
+please see the
+[Vault TOTP backend documentation](/docs/secrets/totp/index.html).
+
+This documentation assumes the TOTP backend is mounted at the
+`/totp` path in Vault. Since it is possible to mount secret backends at
+any location, please update your API calls accordingly.
+
+## Create Key
+
+This endpoint creates or updates a key definition.
+
+| Method   | Path                         | Produces                                                                  |
+| :------- | :--------------------------- | :------------------------------------------------------------------------ |
+| `POST`   | `/totp/keys/:name`           | if generating a key: `200 application/json` else: `204 (empty body)`      |
+
+### Parameters
+
+- `name` `(string: <required>)` – Specifies the name of the key to create. This is specified as part of the URL.
+
+- `generate` `(bool: false)` – Specifies if a key is generated by Vault or if a key is being passed from another service.
+
+- `key_size` `(int: 20)` – Specifies the size in bytes of the Vault generated key. Only used if generate is true.
+
+- `url` `(string: "")` – Specifies the TOTP key url string that can be used to configure a key. Only used if generate is false.
+
+- `key` `(string: <required - if generate is false and url is empty>)` – Specifies the master key used to generate a TOTP code. Only used if generate is false.
+
+- `issuer` `(string: "" <required - if generate is true and url is empty>)` – Specifies the name of the key’s issuing organization.
```

Comment:
Same comment here about generate being true and url is empty not matching the actual code and the description of the parameters in the backend.